### PR TITLE
[PR1] DEV-39 add PacketFactory

### PR DESCRIPTION
**Pull request by jmaralo**
This PR adds PacketFactory to DataTransfer. I've also moved around some files (mainly from domain to infra as dto), but this is subject to change, so I'm open to any suggestions about this decision.

Also I've deleted some commented code.

### CODE REVIEW

**Review by smorsua — CHANGES_REQUESTED** [review]

File: src/data_transfer/infra/dto/packet.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+type id = uint16
+
+type Packet struct {
+	id           id
+	count        uint
+	cycleTime    time.Duration
+	hexValue     []byte
+	measurements map[string]any
```
[the remark this answers is not in the document]
Comment (jmaralo) [reply src/data_transfer/infra/dto/packet.go:16]: We dont have a use for value any more, the only thing it did apart from converting to string was to convert units, which it's now made in a different module

File: src/data_transfer/infra/packet_factory.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+func NewFactory() *PacketFactory {
+	return &PacketFactory{
+		count:     make(map[id]uint),
+		timestamp: make(map[id]time.Time),
+	}
+}
+
+func (factory *PacketFactory) NewPacket(data interfaces.Update) dto.Packet {
```
[the remark this answers is not in the document]
Comment (jmaralo) [reply src/data_transfer/infra/packet_factory.go:24]: But you aren't _getting_ a new packet, you are creating a _new_ packet. That's why I think new is better.

The logic of updating count and cycle time is made by a funcion that returns the next count and cycle time of the packet

File: src/data_transfer/infra/dto/packet.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+func NewPacket(count uint, cycleTime time.Duration, update interfaces.Update) Packet {
```
[the remark this answers is not in the document]
Comment (jmaralo) [reply src/data_transfer/infra/dto/packet.go:19]: Update is an interface, not an struct (just to be clear). I will consider this and tell you later what I think makes the most sense in this case, but I get what you say

File: src/data_transfer/infra/packet_factory.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (smorsua) [reply src/data_transfer/infra/packet_factory.go:24]: Okay, but separate updating the values into another function and then get the updated values from the map

**Review by smorsua — APPROVED** [review]

**Review by lipezaballa — APPROVED** [review]

**Pull request merged**